OpenRA.Game/MiniYaml.cs

```diff
@@ -368,10 +363,15 @@ static List<MiniYamlNode> ResolveInherits(MiniYaml node, Dictionary<string, Mini
 						throw new YamlException(
 							$"{n.Location}: Parent type `{n.Value.Value}` not found");
 
-					if (inherited.TryGetValue(n.Value.Value, out var location))
-						throw new YamlException($"{n.Location}: Parent type `{n.Value.Value}` was already inherited by this yaml tree at {location} (note: may be from a derived tree)");
+					try
+					{
+						inherited = inherited.Add(n.Value.Value, n.Location);
+					}
+					catch (ArgumentException)
+					{
+						throw new YamlException($"{n.Location}: Parent type `{n.Value.Value}` was already inherited by this yaml tree at {inherited[n.Value.Value]} (note: may be from a derived tree)");
+					}
```

Comment:
should use TryAdd here as well

Reply:
Sadly ImmutableDictionary lacks a TryAdd method so this is the best we can do. Thankfully since we're going to throw anyway the performance isn't a concern.